### PR TITLE
Add Npm registry Service Endpoint

### LINKED ISSUE #334: Add Npm registry Service Endpoint
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Hi,

I'd like to configure a Npm registry Service Endpoint in AzureDevops

### New or Affected Resource(s)

* azuredevops_serviceendpoint_npm

### Potential Terraform Configuration

```hcl
resource "azuredevops_serviceendpoint_npm" "example" {
  project_id = azuredevops_project.project.id
  url = "https://registry.npmjs.org"
  access_token = "xxx"
  service_endpoint_name = "example-npm-1"
  description = "example-1"
}
```

### PR DESCRIPTION
**Pull request by pasqualedevita**
## All Submissions:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [X] I have updated the documentation accordingly.
* [X] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [X] My code follows the code style of this project.
* [X] I ran lint checks locally prior to submission.
* [X] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Issue Number: https://github.com/microsoft/terraform-provider-azuredevops/issues/334

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [X] No

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

**Comment by pasqualedevita** [comment]
Test FAILs seems not related to this PR

**Comment by xuzhang3** [comment]
> Test FAILs seems not related to this PR

Integration test, can be ignored.

File: website/docs/r/serviceendpoint_npm.markdown [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  version_control    = "Git"
+  work_item_template = "Agile"
+}
+
+resource "azuredevops_serviceendpoint_npm" "serviceendpoint" {
+
+  project_id            = azuredevops_project.project.id
+  service_endpoint_name = "Sample npm"
+  url                   = "https://registry.npmjs.org"
+  access_token          = "0000000000000000000000000000000000000000"
```
Comment (xuzhang3) [inline website/docs/r/serviceendpoint_npm.markdown]:
```suggestion
  access_token          = "00000000-0000-0000-0000-000000000000"
```

File: azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_npm.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		Description:  "Url for the npm registry",
+	}
+
+	r.Schema["access_token"] = &schema.Schema{
+		Type:             schema.TypeString,
+		Required:         true,
+		Sensitive:        true,
+		DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+		ValidateFunc:     validation.StringIsNotWhiteSpace,
+		Description:      "Authentication token generated through npm (go to Npm > Account > Access Tokens -> Generate New Token)",
```
Comment (xuzhang3) [inline azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_npm.go]: 1. Not all of  the NPM registries have the same navigation style,  suggest remove `(go to Npm > Account > Access Tokens -> Generate New Token)` 
2. Suggest change the description to `The access token for npm registry`

**Review by xuzhang3 — CHANGES_REQUESTED** [review]
Hi @pasqualedevita Thanks for you PR.  Few changes requested.
`serviceendpoint_npm.markdown` should rename to `serviceendpoint_npm.html.markdown` otherwise the document navigation menu can not locate the document.

File: azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_npm.go (hunk 2 above)
Comment (pasqualedevita) [reply azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_npm.go]: I renamed `serviceendpoint_npm.markdown` and changed `access_token` description as suggested.
thank you @xuzhang3

**Comment by pasqualedevita** [comment]
> Hi @pasqualedevita Thanks for you PR. Few changes requested.
> `serviceendpoint_npm.markdown` should rename to `serviceendpoint_npm.html.markdown` otherwise the document navigation menu can not locate the document.

I renamed `serviceendpoint_npm.markdown` and changed `access_token` description as suggested.
thank you @xuzhang3

**Review by xuzhang3 — APPROVED** [review]
LGTM

**Pull request merged**